src/cpp/redis.cpp

```diff
@@ -68,24 +67,21 @@ CommandReply Redis::run(Command& cmd)
             else
                 n_trials = 0;
         }
-        catch (sw::redis::TimeoutError &e) {
+        catch (sw::redis::IoError &e) {
             n_trials--;
             std::this_thread::sleep_for(std::chrono::seconds(2));
         }
-        catch (sw::redis::IoError &e) {
+        catch (sw::redis::ClosedError &e) {
             n_trials--;
             std::this_thread::sleep_for(std::chrono::seconds(2));
         }
-        catch (...) {
+        catch (std::exception& e) {
             n_trials--;
-            throw;
+            throw std::runtime_error(e.what());
```

Comment:
Does it make sense to decrement `n_trials` if we're going to `throw`? I guess it will do no harm, but I wanted to understand if there was some mechanism I don't understand

Reply:
That's a good catch, Al.  I will delete it.